packages/rest/src/rest.server.ts

```diff
@@ -688,6 +710,24 @@ export class RestServer extends Context implements Server, HttpServerLike {
     }
 
     assignRouterSpec(spec, this._externalRoutes.routerSpec);
+
+    if (requestContext) {
```

Comment:
Let's use reverse logic & early return to make the code easier to reason about and keep the level of indentation low.

```ts
if (!requestContext) {
  return spec;
}

if (this.config.openApiSpec.setServersFromRequest)
// ...
return spec;
```

Alternatively, and I think that's a better solution, can you extract this code into a helper function?

```ts
if (requestContext) {
  spec = updateSpecFromRequest(spec, requestContext);
}
return spec;
```